content/docs/guides/authentication/jwt.md

```diff
@@ -276,183 +244,208 @@ Copy the returned JWT string and paste it into the following curl request:
-        response.send(signedJWT)
+        response.send(signedJWT + "\n")
         next()
     }
 
+    let jwtCredentials = CredentialsJWT<ClaimsStandardJWT>(verifier: App.jwtVerifier)
+    let authenticationMiddleware = Credentials()
+    authenticationMiddleware.register(plugin: jwtCredentials)
+    app.router.get("/jwtprotected", middleware: authenticationMiddleware)
     app.router.get("/jwtprotected") { request, response, next in
-        let authHeader = request.headers["Authorization"]
-            guard let authComponents = authHeader?.components(separatedBy: " "),
-                authComponents.count == 2,
-                authComponents[0] == "Bearer",
-                let jwt = try? JWT<ClaimsStandardJWT>(jwtString: authComponents[1], verifier: App.jwtVerifier)
-                else {
-                    let _ = response.send(status: .unauthorized)
-                    return try response.end()
+        guard let userProfile = request.userProfile else {
+                Log.verbose("Failed raw token authentication")
+                response.status(.unauthorized)
+                try response.end()
+                return
             }
-        response.send(jwt)
-        next()
-    }
+        response.send("\(userProfile.id)\n")
+            next()
+        }
 }
 
 extension App {
-    // Example for HMAC signer and verifier
+    // Define JWT signer and verifier here
     static let jwtSigner = JWTSigner.hs256(key: Data("kitura".utf8))
     static let jwtVerifier = JWTVerifier.hs256(key: Data("kitura".utf8))
 }
+
 ```
 
 ---
 
-##Step 9: JWTs on Codable Routes (Optional)
+##Step 9: Using custom claims (Optional)
 
-In our example we used raw routing since we chose to pass the user credentials via the request headers. If we want to use JWTs on our codable routes, we need to encapulate the verification and creation of the users JWT in a TypeSafeMiddleware. We can then register our TypeSafeMiddleware on a Codable route to authenticate the user and access their claims.
+You may want to use your own set of custom claims for your JWT.  For this to work, we need to specify the subject and UserProfileDelegate options.
 
----
+First we will create our claims structure, in our JWTRoutes.swift file:
 
-###Step 9a: Define our type safe middleware.
+```swift
+struct MyClaims: Claims {
+  let id: String
+  let fullName: String
+  let email: String
+}
+```
 
-If you don't already have one, create a Middlewares folder:
+Now we need to edit our UserCredentials model to contain these additional values, so go into your UserCredentials.swift file and add these values to your model:
 
-```
-mkdir Sources/Application/Middlewares
+```swift
+struct UserCredentials: Codable {
+    let username: String
+    let password: String
+    let email: String
+    let fullName: String
+}
 ```
 
-Create a new file, called TypeSafeJWT.swift:
+> For simplicity, we will have the id be the same value as the username in this example.
```

Comment:
For simplicity, in this example, the `id` will have the same value as the `username`.